easybuild/tools/filetools.py

```diff
@@ -1383,22 +1399,8 @@ def path_matches(path, paths):
 def rmtree2(path, n=3):
     """Wrapper around shutil.rmtree to make it more robust when used on NFS mounted file systems."""
 
-    ok = False
-    for i in range(0, n):
-        try:
-            shutil.rmtree(path)
-            ok = True
-            break
-        except OSError as err:
-            _log.debug("Failed to remove path %s with shutil.rmtree at attempt %d: %s" % (path, n, err))
-            time.sleep(2)
-
-            # make sure write permissions are enabled on entire directory
-            adjust_permissions(path, stat.S_IWUSR, add=True, recursive=True)
-    if not ok:
-        raise EasyBuildError("Failed to remove path %s with shutil.rmtree, even after %d attempts.", path, n)
-    else:
-        _log.info("Path %s successfully removed." % path)
+    _log.deprecated("Use 'remove_dir' rather than 'rmtree2'", '5.0')
+    remove_dir(path)
```

Comment:
pass `n` to `remove_dir`? In theory, until `5.0` someone else may still be using `rmtree2` and expecting passing `n` to work...

Reply:
The function removes a directory. The attempts was made an implementation detail when adding `remove_dir`. So it is very unlikely that anyone even notices that another value of n was used. I haven't found any case in EasyBuild that changed the default so there it was not required.

Adding the param to `remove_dir` would mean we have to keep this parameter for a long time and I don't see the gain for that.